src/inventory_ui.cpp

```diff
@@ -252,25 +258,72 @@ void inventory_selector::prepare_paging(itemstack_vector &items)
 void inventory_selector::print_inv_weight_vol(int weight_carried, int vol_carried,
         int vol_capacity) const
 {
-    // Print weight
-    mvwprintw(w_inv, 0, 32, _("Weight (%s): "), weight_units());
-    nc_color weight_color;
-    if (weight_carried > g->u.weight_capacity()) {
-        weight_color = c_red;
-    } else {
-        weight_color = c_ltgray;
+    if(OPTIONS["INV_DISPLAY"] == "list") {
+        // Print weight
+        mvwprintw(w_inv, 0, 32, _("Weight (%s): "), weight_units());
+        nc_color weight_color = c_ltgray;
+        if (weight_carried > g->u.weight_capacity()) {
+            weight_color = c_red;
+        }
+        wprintz(w_inv, weight_color, "%6.1f", convert_weight(weight_carried) + 0.05 ); // +0.05 to round up;
+        wprintz(w_inv, c_ltgray, "/%-6.1f", convert_weight(g->u.weight_capacity()));
+
+        // Print volume
+        mvwprintw(w_inv, 0, 61, _("Volume: "));
+        if (vol_carried > vol_capacity) {
+            wprintz(w_inv, c_red, "%3d", vol_carried);
+        } else {
+            wprintz(w_inv, c_ltgray, "%3d", vol_carried);
+        }
+        wprintw(w_inv, "/%-3d", vol_capacity);
+    } else if(OPTIONS["INV_DISPLAY"] == "grid") {
+        // Print weight
+        mvwprintw(w_inv, 0, TERMX - 41, _("Weight: "));
+        nc_color weight_color = c_ltgray;
+        if (weight_carried > g->u.weight_capacity()) {
+            weight_color = c_red;
+        }
+        wprintz(w_inv, weight_color, "%6.1f", convert_weight(weight_carried) + 0.05 ); // +0.05 to round up;
+        wprintz(w_inv, c_ltgray, "/%-6.1f %s", convert_weight(g->u.weight_capacity()), _(weight_units()));
+        wmove(w_inv, 0, TERMX - 15);
+        wprintz(w_inv, c_ltgray, "[............]");
+        print_bar(0, TERMX - 14, weight_carried, g->u.weight_capacity());
+        
+        // Print volume
+        mvwprintw(w_inv, 1, TERMX - 41, _("Volume: "));
+        wmove(w_inv, 1, TERMX - 30);
+        if (vol_carried > vol_capacity) {
+            wprintz(w_inv, c_red, "%3d", vol_carried);
+        } else {
+            wprintz(w_inv, c_ltgray, "%3d", vol_carried);
+        }
+        wprintw(w_inv, "/%-3d", vol_capacity);
+        wmove(w_inv, 1, TERMX - 15);
+        wprintz(w_inv, c_ltgray, "[............]");
+        print_bar(1, TERMX - 14, vol_carried, vol_capacity);
     }
-    wprintz(w_inv, weight_color, "%6.1f", convert_weight(weight_carried) + 0.05 ); // +0.05 to round up;
-    wprintz(w_inv, c_ltgray, "/%-6.1f", convert_weight(g->u.weight_capacity()));
+}
 
-    // Print volume
-    mvwprintw(w_inv, 0, 61, _("Volume: "));
-    if (vol_carried > vol_capacity) {
-        wprintz(w_inv, c_red, "%3d", vol_carried);
-    } else {
-        wprintz(w_inv, c_ltgray, "%3d", vol_carried);
+void inventory_selector::print_bar(int y, int x, int carried, int capacity) const {
+    wmove(w_inv, y, x);
+    int barsize = static_cast<int>((carried * 24.0) / capacity);
```

Comment:
Can `vol_capacity` can be 0 (for a naked character), this should be handled here.